spec/datadog/tracing/contrib/kafka/data_streams_spec.rb

```
@@ -0,0 +1,279 @@
+require 'datadog/core/ddsketch'
+require 'ostruct'
+require 'datadog/tracing/contrib/kafka/integration'
+require 'datadog/tracing/contrib/kafka/instrumentation/producer'
+require 'datadog/tracing/contrib/kafka/instrumentation/consumer'
+module ActiveSupport
+  module Notifications
+    def self.subscribe(*_args)
+    end
+  end
+end
+
+# Mock Kafka classes that we need
+module Kafka
+  class Producer
+    attr_accessor :pending_message_queue
+
+    def initialize
+      @pending_message_queue = []
+    end
+
+    def deliver_messages(**kwargs)
+      result = {delivered_count: @pending_message_queue.size}
+      @pending_message_queue.clear
+      result
+    end
+
+    def send_messages(messages, **kwargs)
+      {sent_count: messages.size}
+    end
+  end
+
+  class Consumer
+    def each_message(**kwargs)
+      if block_given?
+        yield OpenStruct.new(
+          topic: 'test_topic',
+          partition: 0,
+          offset: 100,
+          headers: {}
+        )
+      end
+    end
+
+    def each_batch(**kwargs)
+      if block_given?
+        yield OpenStruct.new(
+          topic: 'test_topic',
+          partition: 0,
+          messages: [
+            OpenStruct.new(offset: 100, key: 'key1'),
+            OpenStruct.new(offset: 101, key: 'key2')
+          ]
+        )
+      end
+    end
+  end
+end
+require 'datadog'
+
+RSpec.describe 'Kafka Data Streams instrumentation' do
+  let(:configuration_options) { {} }
+
+  before do
+    Datadog.configure do |c|
+      c.tracing.instrument :kafka, configuration_options
+      c.data_streams.enabled = true
+    end
+  end
+
+  around do |example|
+    # Reset before and after each example; don't allow global state to linger.
+    Datadog.registry[:kafka].reset_configuration!
+    example.run
+    Datadog.registry[:kafka].reset_configuration!
+  end
+
+  describe 'pathway context' do
+    before do
+      skip('DDSketch not available') unless Datadog::Core::DDSketch.supported?
```

Is it possible for ddsketch to fail to build or fail to be loaded and then all of the tests being silently skipped, thus providing no feedback that the feature isn't tested and possibly is not working?

---

+1 this is an extremely good point.

I suggest perhaps stealing [from profiling's `skip_if_profiling_not_supported`](https://github.com/DataDog/dd-trace-rb/blob/master/spec/datadog/profiling/spec_helper.rb#L21) that skips on jruby + truffleruby + by default on macOS but otherwise assumes that the library must be up and running and fails the test if that's not the case.

This helps avoid the "oops it was all green because we skipped all the tests"

---

thanks for this. turns out i was skipping a bunch of tests